handout/http/requests_as_promises.md

```diff
@@ -0,0 +1,40 @@
+## Requests as Promises
+
+Although the Angular http client returns an observable you can also convert it into a promise.
+
+```ts
+import { Http } from '@angular/http';
+import { Injectable } from '@angular/core';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/toPromise';
+
+@Injectable()
+export class SearchService {
+
+  constructor(private http: Http) {}
+
+  search(term: string) {
+    return this.http
+      .get('https://api.spotify.com/v1/search?q=' + term + '&type=artist')
+      .map((response) => response.json())
+      .toPromise();
+  }
+}
+```
+
+We would then consume it as a regular promise in the component.
+
+```ts
+@Component({ ... })
+export class MyApp {
+	...
+
+	search() {
+		this.searchService.search(this.searchField.value)
+		  .then((result) => {
+        this.result = result.artists.items;
+      })
+      .catch((error) => console.error(error));
+	}
+}
+```
```

Comment:
I bet many readers will wonder which approach to follow, should I use the observable or the promise pattern? Some of them, newer to Angular 2, would be inclined to always call `.toPromise` to have something familiar to work with but we should warn them about the downside of doing this: they will be losing the ability to cancel a request and the ability to chain additional rxjs operators to further transform the stream (in the component consuming the service for example).